### ISSUE TITLE
SMEs - Please add your draft review comments here

### ISSUE THREAD

**Issue by DawnmarieDesJardins** [solliancenet/MCW-Predictive-Maintenance-for-Remote-Field-Devices#7]
Hello, 
As you review this draft, please add all comments below.  This will make sure everyone sees all feedback and makes conversation easier.

We appreciate your time and expertise!

**Comment by olivierbloch** [comment]
Main README:
- I noticed several typos here and there but I assume that they will be fixed in upcoming passes.
- I think we need to make it clearer that the workshop is about predictive maintenance in the cloud as opposed to at the Edge. It is never said explicitly that we will be running ML algorithms in the cloud and not at the Edge (which for this kind of scenario is usually preferred, but we want to focus on the predictive maintenance part of it). May be we can mention the Edge intelligence topic at the end of the workshop with follow up resources (or workshop) to bring ML at the Edge

**Comment by olivierbloch** [comment]
Workshop:
- Why are we not using VS Code instead of VS. It seems that having to have Visual Studio 2019 simply to run a simulator is an overkill...
- I think we should invite user to navigate the simulated rod pump device code and highlight the parts where device is sending telemetry (to show how data is formatted for IoT Central) and where device receives commands
- Steps 4, 5 and 6 are not in there at all... did I miss something? We are skipping the whole continuous export and Machine Learning part.

**Comment by joelhulen** [comment]
Thanks for the feedback, @olivierbloch. The main readme will be updated once finalized. The primary content we should focus on for the review is in the Whiteboard Design Session - Trainer guide, specifically, and the Hands-on lab. The final components of the lab will be completed tomorrow, which is primarily around creating a custom dashboard and lab cleanup.

Per your feedback:

* Please read the WDS as opposed to the main readme.
* Please read below this bulleted list on the ML components.
* We can switch to VS Code.
* We'll walk users through the important parts of the code within the lab.
* Steps 4, 5, and 6 will be completed tomorrow.

I sent the following in an email on July 19th, regarding the ML components. Could you please read through and reply? We ended up removing the ML portion from the package due to lack of response and clear direction:

----------------------------------------------------

> I am finalizing the architecture for the MCW and wanted to clarify some things. I understand that the primary focus is on IoT Central, and the benefits of using SaaS over PaaS. As you know, this somewhat limits any customization we can do without adding a host of additional Azure services. This leads me to the following questions and my proposed solution. I would like to see if you agree.

> One of the scenario requirements is to enable alerts within IoT Central and to send out notifications using Flow. We have a trained machine learning model we created to predict impending pump failure based on telemetry. To use this trained model for near-real-time scoring, we have two different options we’ve considered:

> 1. Continuously export telemetry to Event Hubs, trigger an Azure Function to send telemetry to a web service that hosts the model for scoring, then trigger Microsoft Flow to send an alert. We could not see a way to add an alert to IoT Central from the function. This option requires Event Hubs and Azure Functions, along with custom code.
> 2. Create an IoT Edge device and load the trained model onto it for scoring at the edge. If a failure is predicted, send a failure prediction event that the attendees can create a trigger for in the lab. The benefit of this is that there is no need to add additional services.

> Which of these options seem better to you?

> Am I correct that for this MCW, we want to limit external components as much as possible so we can focus on IoT Central and the benefits it provides by being a SaaS product?

----------------------------------------------------

I sent a follow-up email later on the 19th in response to Vincent Hong who said he did not believe IoT Edge devices are compatible with IoT Central:

----------------------------------------------------

> From what I understood, I thought you could connect an IoT Edge device to IoT Central and send telemetry in the same way as you would directly from a device. In fact, the IoT Central docs states the following:

> From: https://docs.microsoft.com/en-us/azure/iot-central/concepts-connectivity

> _If your device can't use any of the supported protocols, you can use Azure IoT Edge to do protocol conversion. IoT Edge supports other intelligence-on-the-edge scenarios to offload processing to the edge from the Azure IoT Central application._

> Upon further investigation, it seems like you are correct in that there’s no way to directly register an IoT Edge device. I found this repo for a bridge for connecting IoT Edge devices to IoT Central. Unfortunately, I think it requires you to set up a separate IoT Hub to register the edge device, then send the telemetry to IoT Central. The other limitation is that communication is unidirectional from the edge device to IoT Central. It cannot receive commands and forward to connected IoT devices: https://github.com/iotblackbelt/EdgeIOTCBridgeSolution

> Given these limitations, it seems that using IoT Edge in the way I had hoped is dead in the water. Are there any known workarounds for getting IoT Edge devices connected to IoT Central, or should we go with option 1 below?

----------------------------------------------------

Due to lack of direction from the team after sending the above two emails, we have created a strictly IoT Central-focused MCW package. We do have a custom ML model we created that makes predictions based on training data we generated. The primary question is whether this MCW is the right one to add in the ML components, or if we would be better served creating another MCW that focuses more on ML along with IoT PaaS services?

**Comment by olivierbloch** [comment]
To the question about where to run the ML algorithm, I think we need to keep it in the Cloud as Edge is not yet supported in IoT Central and it would confuse people to have to go through hoops to make Edge work with IoT Central.
For the best way to implement ML behind IoT Central, we would need to involve a couple more folks to the discussion to go into details of what’s the best way to do so as there are several options (going through Event Hubs as described below is one, going through blob storage is another). For the alert back to IoT Central, at this point you can only send a command (Direct Method) to a device using the Microsoft Flow connector. Sending a notification out (not in IoT Central) seems reasonable though.
A simpler way to implement anomaly detection in the cloud is to leverage Azure Stream Analytics as described in this blog post<https://azure.microsoft.com/en-us/blog/anomaly-detection-using-built-in-machine-learning-models-in-azure-stream-analytics/>. It abstract the complexity of having to host a web app to run the model. I suggest we go this route (IoT Central -> Event Hub -> Stream Analytics with ML -> MSFT Flow



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by joelhulen** [comment]
Given your suggestion of using IoT Central -> Event Hub -> Stream Analytics with ML -> MSFT Flow, how would you like to proceed? Should we wait to involve others for feedback? This approach will require a change to how we're tracking and alerting on anomalies in the hands-on-lab, where the process is done entirely in IoT Central. Should we keep the lab as-is and add your suggested flow to the WDS? The WDS can be expanded to discuss this method of anomaly detection and a more advanced option that uses a custom trained model. I did not add these ML components in the WDS because I wasn't sure how they fit in with the expressed desire to focus on SaaS and IoT Central, specifically.

**Comment by olivierbloch** [comment]
As of now the process doesn’t detect anomalies, it triggers an alert when a threshold is set manually. There is no Machine Learning Anomaly detection involved. The very promise of the lab is to do predictive maintenance, which by definition involves machine learning
Do you want to jump on a call tomorrow to discuss further and close on this?


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by joelhulen** [comment]
Agreed with your assessment. We even have a custom-trained model we built for this lab. However, the approach we wanted to take with it bypassed a lot of IoT Central's capabilities, which is why we posed the questions we did. I would like to discuss this in a call so we can nail down the ideal approach, with your goals in mind.
